Feat/custom

baraa0abd opened the pull request:
## What Changed
- Implement Custom-your-prod screen
- nav from prod img to customize the prod

## Screenshots/Videos (if applicable)


https://github.com/user-attachments/assets/a686a6f9-5450-42b7-9cf6-0b47be13e40e



## Checklist
- [ ] My code follows the project's style guidelines
- [ ] Changes have been tested manually and verified.
- [ ] PR includes at most one single feature.

baraa0abd committed:
feat: Add 'Design Editor' screen and refactor Home

- Introduces a new `DesignEditorScreen` for product customization.
- The new screen is composed of several widgets:
    - `ProductImage`: Displays the main product image.
    - `PriceQuantityRow`: Manages product price and quantity selection.
    - `ColorSelector`: Allows users to pick a product color.
    - `SizeSelector`: Enables users to choose a size.
    - `UploadLogoSection`: Provides an interface for uploading a logo.
    - `BottomButtons`: Contains 'Add to cart' and 'Reset' actions.
- Adds the `intl` package dependency.
- Removes the "Top Stores" section from the `HomeScreen`.

baraa0abd committed:
feat: Implement design editor screen and UI updates

- Adds a new `DesignEditorScreen` accessible by tapping a product in the `GridProduct` widget.
- Replaces the PNG product image with an optimized WEBP version.
- Introduces a new SVG icon (`ic_uplod.svg`) for the upload action.
- Updates the `UploadLogoSection` to use the new SVG icon instead of the default `Icons.upload_file_outlined`.
- Refactors hardcoded primary colors in `SizeSelector` and `BottomButtons` to use the `SellioColors` theme constants.

baraa0abd committed:
fix(ui): Adjust font weight for "Upload logo" title

- Changes the `fontWeight` of the "Upload logo or image" text from `w600` to `w500`.

baraa0abd committed:
refactor(ui): Update button styles in customize product screen

Commit:
Branch was auto-updated.

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.

israamohamed107 reviewed (changes requested):
Good work Baraa, Just one small thing to adjust - the selected color styling doesn't match the Figma design.
When a color is selected, the circle should become larger with a surrounding border circle. 

<img width="401" height="188" alt="image" src="https://github.com/user-attachments/assets/ec60c8d8-ac72-42d2-9a1a-17f44181b6ae" />

mohamedebrahem13 reviewed (commented):
The old price has horizontal line this line disappeared when you scroll up or down

baraa0abd committed:
Merge remote-tracking branch 'origin/develop' into feat/custom

baraa0abd committed:
refactor(ui): Update styles for color and size selectors

- Changes the font weight of the 'Size' title from 600 to 500.
- Refactors the `ColorSelector` to add a border around the selected color, matching the updated design.
- Adjusts the size of the color circles from 40x40 to 36x36.

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.

baraa0abd committed:
feat(ui): Enhance price display in `PriceQuantityRow`

- Modifies the styling of the old price to use a thicker strikethrough and a hint color for better visual distinction.
- Adjusts the current price style, setting the color to the primary theme color and changing the font weight.

baraa0abd commented:
> Good work Baraa, Just one small thing to adjust - the selected color styling doesn't match the Figma design. When a color is selected, the circle should become larger with a surrounding border circle.
> 
> <img alt="image" width="401" height="188" src="https://private-user-images.githubusercontent.com/74803351/511604177-ec60c8d8-ac72-42d2-9a1a-17f44181b6ae.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjI1OTY2NDIsIm5iZiI6MTc2MjU5NjM0MiwicGF0aCI6Ii83NDgwMzM1MS81MTE2MDQxNzctZWM2MGM4ZDgtYWM3Mi00MmQyLTlhMWEtMTdmNDQxODFiNmFlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTA4VDEwMDU0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk5MGRjZTc5Njc5Njc3ZTBmM2JhNGVjMjlmMzM3YjBhZDcxNDJhMzE0NTY4YTZmYmY2MDQ4MGU2NmVjMzIzMjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.FqQdZILj_HfXRLLoGVhTH2Pc67cKwwg47cT5v6EscRk">

I think it’s actually smaller, but the border makes it look bigger than the others

Commit:
Branch was auto-updated.

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.

baraa0abd committed:
chore: Format price_quantity_row.dart

Applied auto-formatting to the `price_quantity_row.dart` widget file for better code readability and consistency. No functional changes were made.

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.

baraa0abd committed:
feat: Implement image picker for logo upload

- Adds the `image_picker` dependency to `pubspec.yaml`.
- Implements functionality to pick an image from the gallery in the `UploadLogoSection`.
- Displays the selected image preview on the screen.
- Adds necessary Android permissions for camera and storage access (`AndroidManifest.xml`).
- Includes a `FileProvider` configuration for handling file URIs (`provider_paths.xml`).

baraa0abd committed:
feat: Overlay uploaded image on product

- Adds the ability to display an uploaded image as an overlay on the main product image.
- Creates a `ProductImage` widget that takes an optional `File` (`overlayImage`) and displays it on top of a base product asset.
- Refactors the `UploadLogoSection` to use the new `ProductImage` widget, showing a live preview of the selected logo on the product.
- Adjusts the layout of the upload area to show the uploaded image as a preview instead of within the upload box itself.

baraa0abd committed:
feat: Implement image upload for product customization

- Adds functionality to upload a logo or image from the gallery on the `DesignEditorScreen`.
- The selected image is now overlaid onto the main product image.
- State management for the selected image is handled in the parent `DesignEditorScreen`.
- The 'Reset' button now clears the selected image in addition to other options.
- The `UploadLogoSection` widget is updated to display the chosen image and pass the selected file up to the parent screen.

baraa0abd committed:
Merge remote-tracking branch 'origin/develop' into feat/custom

# Conflicts:
#	lib/core/app_management/route/route_manager.dart
#	lib/ui/screens/home/home_screen.dart

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.
